File: 299. Bulls and Cows/solution.js
/**
 * @param {string} secret
 * @param {string} guess
 * @return {string}
 */
var getHint = function(secret, guess) {
  let a = 0;
  let b = 0;
  const length = guess.length;
  const secs = new Array(10).fill(0);

  for (let i = 0; i < length; i++) {
    const s = secret[i];
    const g = guess[i];
    if (g === s) {
      a++;
    } else {
      if (secs[s]++ < 0) b++;
      if (secs[g]-- > 0) b++;
    }
  }

  return `${a}A${b}B`;
};

File: 299. Bulls and Cows/solution.py
from collections import Counter


class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        bull = sum(map(operator.eq, secret, guess))
        a = Counter(secret)
        b = Counter(guess)
        cow = sum((a & b).values()) - bull
        return '{}A{}B'.format(bull, cow)
